src/gui/log/logview.h

```diff
@@ -26,35 +26,34 @@
  * exception statement from your version.
  */
 
-#ifndef LOGLISTWIDGET_H
-#define LOGLISTWIDGET_H
+#pragma once
+
+#include <QTreeView>
 
-#include <QListWidget>
 #include "base/logger.h"
 
-class QKeyEvent;
+class LogModel;
+class LogFilterModel;
 
-class LogListWidget : public QListWidget
+class LogView : public QTreeView
 {
     Q_OBJECT
+    Q_DISABLE_COPY(LogView)
 
 public:
-    // -1 is the portable way to have all the bits set
-    explicit LogListWidget(int maxLines, const Log::MsgTypes &types = Log::ALL, QWidget *parent = nullptr);
-    void showMsgTypes(const Log::MsgTypes &types);
-
-public slots:
-    void appendLine(const QString &line, const Log::MsgType &type);
+    LogView(const QList<QPair<QString, Log::MsgType>> &logLines, const Log::MsgTypes types, QWidget *parent = nullptr);
+    void setMsgTypes(const Log::MsgTypes types);
+    void appendLine(const QString &line, const Log::MsgType &type = Log::NORMAL);
```

Comment:
and here `Log::MsgType type = Log::NORMAL`

Reply:
Irrelevant in new code